binary_gentoo/internal/cli/tree_diff.py

```diff
@@ -22,11 +37,32 @@ def iterate_new_and_changed_ebuilds(config):
         for ebuild_file in ebuild_files:
             old_portdir_ebuild_file = os.path.join(config.old_portdir, category_plus_package,
                                                    ebuild_file)
+            new_portdir_ebuild_file = os.path.join(root, ebuild_file)
+
+            # don't output if files are identical
             if os.path.exists(old_portdir_ebuild_file):
-                new_portdir_ebuild_file = os.path.join(root, ebuild_file)
                 if filecmp.cmp(old_portdir_ebuild_file, new_portdir_ebuild_file):
                     continue
 
+            if config.keywords:
+                with open(new_portdir_ebuild_file, "r") as ifile:
+                    new_ebuild_content = ifile.read()
+                new_ebuild_relevant_keywords = _get_relevant_keywords_set_for(new_ebuild_content, config.keywords)
+
+                # don't output if the new file doesn't include the specified keywords
+                if len(new_ebuild_relevant_keywords) == 0:
+                    continue
+
+                if os.path.exists(old_portdir_ebuild_file):
+                    with open(old_portdir_ebuild_file, "r") as ifile:
+                        old_ebuild_content = ifile.read()
+                    old_ebuild_relevant_keywords = _get_relevant_keywords_set_for(old_ebuild_content, config.keywords)
+
+                    # don't output if both old and new file include the same keywords
+                    # (i.e., when only other keywords have changed)
+                    if new_ebuild_relevant_keywords == old_ebuild_relevant_keywords:
+                        continue
```

Comment:
I'm not sure that last shortcut is a good idea. There could be changes in non-keyword lines of the ebuild, let's drop this check, no?

With that check gone, function `_get_relevant_keywords_set_for` could be simplified to return a bool rather than a set and be called e.g. `_matches_included_keywords` or so.

Reply:
I have taken the assumption that for non-trivial changes which need to cause a re-emerge, the version of the ebuild would be incremented instead of making modifications to an existing version. If that assumption is invalid, then indeed this should be dropped.

Reply:
There is a rule that Gentoo packagers need to do a "rev bump" (e.g. `-r2` to `-r3`) when the content of the package or its runtime dependencies change, yes. However sometimes rev bumps are forgotten. The point of the current `filecmp` code in there was to be robust towards those cases, at the risk of building too much (ignoring `--shy-rebuild` for the moment). If you consider that an anti-feature personally, we can make it a command line flag. What do you think?

Reply:
Do you know how portage itself treats this? If a package changes but the version/revision remains the same, will something like `emerge --update --deep @world` do an update or will it not do anything at all? As with some of the other discussions we have had, I would go for the same default behavior as portage itself.

That being said, personally I think it is interesting to give the user a command line flag to select the opposite behavior too.

Reply:
`--dynamic-deps` comes into mind here. It seems like `--dynamic-deps=y` has become default by now, which could cause some action despite a missing revbump, I think. I can see value in both version — ignoring content changes or reporting it — so a new flag seems like a nice idea. I have `--eager` or `--pessimistic` in mind for a name, but those are probably more fun than self-explaining. Maybe `--report-changes` to report files that changed and have the default not report them? (I notice we'd probably still want to not report changes if the only line that changed was the keyword one, but we can keep that for later.)

Reply:
That seems good to me.